File: TheLoversApi/src/main/java/com/strilicherk/theloversapi/feature_user/domain/dto/UserResponseDTO.java
package com.strilicherk.theloversapi.feature_user.domain.dto;

import com.strilicherk.theloversapi.core.domain.model.user_location.UserLocationRequestDTO;

import java.time.LocalDate;
import java.util.UUID;

public record UserResponseDTO(
        UUID id,
        String name,
        String email,
        String phone,
        Boolean userVerified,
        LocalDate birthDate,
        Integer genderId,
        UserLocationRequestDTO userLocation,
        Integer roleId
) {
}

File: TheLoversMobileWeb/composeApp/src/commonMain/kotlin/org/example/thelovers/feature_user/data/dto/UserResponseDTO.kt
package org.example.thelovers.feature_user.data.dto

import kotlinx.serialization.Serializable
import org.example.thelovers.core.domain.model.Gender
import org.example.thelovers.feature_location.data.LocationResponseDTO

@Serializable
data class UserResponseDTO(
    val id: String,
    val name: String?,
    val email: String?,
    val phone: String,
    val userVerified: Boolean,
    val birthDate: String?,
    val genderId: Int?,
    val userLocation: LocationResponseDTO?,
    val roleId: Int,
    val createdAt: String,
    val updatedAt: String
) {
    fun isIncomplete(): Boolean {
        if (name.isNullOrBlank()) return true
        if (birthDate == null) return true
        if (userLocation != null && userLocation.isIncomplete()) return true
        if (genderId != null && Gender.fromId(genderId) == null) return true
        return false
    }
}






